src/mixins/richEditor/index.js

```diff
@@ -145,9 +151,10 @@ export default {
 			wrapper.appendChild(mount)
 			document.body.appendChild(wrapper)
 
-			// Mount and get raw html
+			// Mount and get raw HTML. As HTML replaces new lines, empty lines and consecutive whitespaces
+			// with a single whitespace, which parseContent() can't trim, we do it manually
 			Item.$mount(mount)
-			const renderedHtml = wrapper.innerHTML
+			const renderedHtml = wrapper.innerHTML.replace(/>\s+</g, '><')
```

Comment:
It seems that the problem is not in the component rendering, but in `parseContent` where HTML is transformed to text (though it probably doesn't hurt, for component rendering to HTML, it is valid to have these spaces).

Reply:
When HTML is transformed, `stripTags` omits everything except for `@userId` and these spaces laying between tags. As user id could also contain spaces, I'm not sure if it would be better to use `replace` here, or match `parseContent` result with a regex again

Reply:
> When HTML is transformed, `stripTags` omits everything except for `@userId` and these spaces laying between tags.

Yes, removing repeating spaces between tags should be before stripping tags. Same as here — for HTML. The problem in `NcRichContenteditable` is that then it takes value from content, it takes it from HTML content, and getting text content from HTML is invalid, 

`parseContent` works incorrectly — it takes not correct text content of HTML, not what native `textContent` would take.

While `renderComponent` works correctly, the returned HTML content is valid as Vue rendering is valid.

For this issue, it works both ways. But we cannot guarantee that we will never have HTML in contenteditable not from `renderComponent` or that spaces in `renderComponent` would be needed in some other place.

> As user id could also contain spaces, I'm not sure if it would be better to use `replace` here

But it cannot contain `/>\s+</`. Otherwise, changing them here wouldn't be safe as well.

Reply:
I couldn't find an easy solution for this atm, so will drop this commit in favor of merging user regex fix

Reply:
What I'm proposing is to just move your solution from from `renderComponent` to `parseContent`.

In the beginning of `parseContent`:
```diff
- let text = content.replace(/<br>/gmi, '\n')
- text = text.replace(/&nbsp;/gmi, ' ')
+ let text = content.replace(/>\s+</g, '><')
+ text = text.replace(/<br>/gmi, '\n')
+ text = text.replace(/&nbsp;/gmi, ' ')
```

Because this is `parseContent` - the method that doesn't parse HTML correctly, not `renderComponent` that renders HTML incorrectly.

Reply:
![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/55860977-a27c-4d9f-8675-f96acfb5a223)

Reply:
There's also a check in NcRichcontenteditable
`this.value.trim() === this.parsedContent(htmlOrText)`. I'm not sure, whether it should be also trimmed by that moment or not

Reply:
I have only found check `this.value.trim() !== this.parseContent(html).trim()` in watcher for value. It updates internal value and content based on value update. But I don't see how it is related to rendering/parsing. Both value and content from html either have spaces in between or not. When it is supposed to be equal, value is got from the same `this.parseContent` in `updateValue`.